magda-authorization-api/src/test/buildCreateApiRouter.spec.ts

```diff
@@ -0,0 +1,505 @@
+import {} from "mocha";
+import * as sinon from "sinon";
+import * as request from "supertest";
+import * as express from "express";
+import * as yargs from "yargs";
+import addJwtSecretFromEnvVar from "@magda/typescript-common/dist/session/addJwtSecretFromEnvVar";
+import buildJwt from "@magda/typescript-common/dist/session/buildJwt";
+import createApiRouter from "../createApiRouter";
+import { expect } from "chai";
+import jsc from "@magda/typescript-common/dist/test/jsverify";
+import mockDatabase from "./mockDatabase";
+import mockUserDataStore from "@magda/typescript-common/dist/test/mockUserDataStore";
```

Comment:
Because this has state and is a singleton, it's possible that one test will leave it a state that breaks the next one. Can you make it so that this is recreated or reset between individual tests?

Reply:
I thought it might be a more complete test if we test the whole loop --- i.e. first test create a user and the next test can still pull it out later. Thus, kept the state. But I think one thing you've mentioned could be right --- we should not make tests depends on each other. have to admit I still want to save some effort here on unit tests ;-)  Will re-structure the test later.

Reply:
this one has been done~